docs: update example with correct field names

dgwhited opened the pull request:
### Background

Updates the example with the field names required by the provider

### Changes

* updates fields

### Testing

* tested with a tf plan

danbiwer commented:
Hi Dustin, thanks for catching that!

This is failing because the doc that you are modifying is generated from [examples/resources/panther_s3_source/resource.tf](https://github.com/panther-labs/terraform-provider-panther/blob/main/examples/resources/panther_s3_source/resource.tf)

After modifying the file in the examples directory, the docs can be regenerated by running `go generate`

dgwhited commented:
> Hi Dustin, thanks for catching that!
> 
> This is failing because the doc that you are modifying is generated from [examples/resources/panther_s3_source/resource.tf](https://github.com/panther-labs/terraform-provider-panther/blob/main/examples/resources/panther_s3_source/resource.tf)
> 
> After modifying the file in the examples directory, the docs can be regenerated by running `go generate`

Hi Dan, my mistake! I updated the `resource.tf` file, ran `go generate`, and updated the PR with that commit.
